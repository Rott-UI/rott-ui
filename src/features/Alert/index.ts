export * from './component'
export * from './models'
