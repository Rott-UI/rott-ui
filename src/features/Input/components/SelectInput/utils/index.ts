export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
