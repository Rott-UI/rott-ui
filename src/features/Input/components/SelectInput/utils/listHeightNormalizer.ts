/* eslint-disable curly */
// Component Imports
import {ITEM_HEIGHT, LIST_MAX_ITEM_COUNT, LIST_MIN_ITEM_COUNT, SEPARATOR_HEIGHT} from '../constants'

/**
 * Liste Yuksekligini Tanimlanan Min-Max Item Degerlerine Gore Hesaplar.
 * @param itemCount eleman sayisi
 * @returns liste yuksekligi
 */
export const listHeightNormalizer = (itemCount: number, showDescription: boolean) => {
  if (itemCount < LIST_MIN_ITEM_COUNT)
    return (
      ITEM_HEIGHT(showDescription) * LIST_MIN_ITEM_COUNT +
      SEPARATOR_HEIGHT * (LIST_MIN_ITEM_COUNT + 1)
    )
  else if (itemCount >= LIST_MAX_ITEM_COUNT)
    return (
      ITEM_HEIGHT(showDescription) * LIST_MAX_ITEM_COUNT +
      SEPARATOR_HEIGHT * (LIST_MAX_ITEM_COUNT + 1)
    )
  else return ITEM_HEIGHT(showDescription) * itemCount + SEPARATOR_HEIGHT * (itemCount + 1)
}
