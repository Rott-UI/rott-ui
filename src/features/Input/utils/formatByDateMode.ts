/* eslint-disable indent */
// Util and Lib Imports
import {
  TranslationLanguageTypes,
  formatDate,
  formatDateTime,
  formatMessage,
  formatTime,
} from '../../../utils'

export const formatByDateMode = (
  mode: 'date' | 'time' | 'datetime' | 'modal-date' | 'modal-time' | 'modal-datetime',
  date: string
) => {
  switch (mode) {
    case 'time':
      return formatTime(date)
    case 'date':
      return formatDate(date)
    default:
      return formatDateTime(date)
  }
}

// TODO: Genisletilecek - Refactor Edilecek
/**
 *
 * @param date YYYY.MM.DD formatinda tarih
 * @param mode donecek format tipi
 * @returns
 * mode: **'DD MM YYYY'** - ```{day: DD, month: MM, year: YYYY}```
 *
 * mode: **'DD MMMM YYYY'** - ```{day: DD, month: 'Month Text', year: YYYY}```
 */
export const dateFormatSlicer = (date: string, mode: 'DD MM YYYY' | 'DD MMMM YYYY') => {
  const dateArr = date.split('.') //YYYY.MM.DD

  if (dateArr.length === 3) {
    if (mode === 'DD MM YYYY') return {day: dateArr[2], month: dateArr[1], year: dateArr[0]}
    else {
      return {
        day: dateArr[2],
        month: formatMessage(`DATE.MONTH.${dateArr[1]}` as TranslationLanguageTypes),
        year: dateArr[0],
      }
    }
  }

  return null
}

export const dateYearFormatter = (date: string) => {
  const transferDate = new Date(date)

  const formattedDate = `${transferDate.getDate().toString().padStart(2, '0')}.${(transferDate.getMonth() + 1).toString().padStart(2, '0')}.${transferDate.getFullYear()}`

  return formattedDate
}

/**
 * @param date DD / MM / YYYY formatinda tarih
 * @returns
 */
export const dateYearConfirmationFormatter = (date: string) => {
  const transferDate = new Date(date)

  const formattedDate = `${transferDate.getDate().toString().padStart(2, '0')} / ${(transferDate.getMonth() + 1).toString().padStart(2, '0')} / ${transferDate.getFullYear()}`

  return formattedDate
}
