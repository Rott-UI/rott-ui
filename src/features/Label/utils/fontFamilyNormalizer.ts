import {FontWeight} from '../models'

export const fontFamilyNormalizer = (weight: FontWeight = 500) => {
  switch (weight) {
    case 100:
    case 200:
    case 300:
      return 'Markpro-ExtraLight'
    case 400:
      return 'Markpro-Light'
    case 500:
      return 'Markpro-Book'
    case 600:
    case 'normal':
      return 'Markpro-Medium'
    case 700:
    case 'bold':
      return 'Markpro-Bold'
    case 800:
      return 'Markpro-Heavy'
    case 900:
      return 'Markpro-Black'
    default:
      return 'Markpro-Medium'
  }
}
