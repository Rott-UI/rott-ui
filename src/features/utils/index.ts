export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
