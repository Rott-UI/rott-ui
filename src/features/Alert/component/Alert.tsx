// React Imports
import React, {FC} from 'react'

// Feature Imports
import {Icon} from '../../Icon'
import {Item} from '../../Item'
import {Label} from '../../Label'
import {AlertModel} from '../models'

// Util and Lib Imports
import {colorFromVariant} from '../../../utils'

export const Alert: FC<AlertModel> = ({text, size, variant, leftIcon, rightIcon}) => {
  return (
    <Item
      testID='alert-test-id'
      row
      size={size}
      backgroundColor={colorFromVariant(variant)}
      paddingHorizontal={16}
      paddingVertical={8}
      alignItemsCenter>
      {leftIcon && (
        <Icon
          testID='alert-left-icon'
          width={24}
          height={24}
          mode='stroke'
          strokeWidth={1.5}
          variant='white'
          marginRight={8}
          {...leftIcon}
        />
      )}

      <Label
        width={330}
        height={16}
        variant='white'
        size='sm'
        {...(typeof text === 'object' ? {...text} : null)}>
        {typeof text === 'string' ? text : text.text}
      </Label>

      {rightIcon && (
        <Icon
          testID='alert-right-icon'
          width={24}
          height={24}
          mode='stroke'
          strokeWidth={1.5}
          variant='white'
          marginRight={8}
          {...rightIcon}
        />
      )}
    </Item>
  )
}
