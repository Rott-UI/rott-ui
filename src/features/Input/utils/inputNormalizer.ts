/* eslint-disable indent */

// Model Imports
import {Size} from '../../../models'

// Provider Imports
import {themeConfig} from '../../../providers'

// Util and Lib Imports
import {display} from '../../../utils'

interface InputStyleProps {
  height: number
  paddingHorizontal: number
  placeholderSize: number
  bottomElementPadding: number
  icon: {
    height: number
    width: number
    paddingBottom: number
  }
}

interface InputStyleNormalizerProps {
  size?: Size
  placeholderSize?: Keyof<typeof themeConfig.fontSizes> | number
}

export function InputStyleNormalizer({
  size = 'lg',
  placeholderSize,
}: InputStyleNormalizerProps): InputStyleProps {
  switch (size) {
    case 'xs':
    case 'sm':
      return {
        height: display.normalize(40, 'height'),
        paddingHorizontal: display.normalize(16, 'height'),
        placeholderSize:
          typeof placeholderSize === 'number' ? placeholderSize : themeConfig.fontSizes.sm,
        bottomElementPadding: display.normalize(6, 'height'),
        icon: {
          height: 18,
          width: 18,
          paddingBottom: 8,
        },
      }
    case 'lg':
    case 'xl':
    case '2xl':
    case 'full':
      return {
        height: display.normalize(56, 'height'),
        paddingHorizontal: display.normalize(16, 'height'),
        placeholderSize:
          typeof placeholderSize === 'number' ? placeholderSize : themeConfig.fontSizes.xl,
        bottomElementPadding: display.normalize(12, 'height'),
        icon: {
          height: 24,
          width: 24,
          paddingBottom: 16,
        },
      }

    case 'md':
    default:
      return {
        height: display.normalize(48, 'height'),
        paddingHorizontal: display.normalize(16, 'height'),
        placeholderSize:
          typeof placeholderSize === 'number' ? placeholderSize : themeConfig.fontSizes.lg,
        bottomElementPadding: display.normalize(8, 'height'),
        icon: {
          height: 24,
          width: 24,
          paddingBottom: 12,
        },
      }
  }
}
