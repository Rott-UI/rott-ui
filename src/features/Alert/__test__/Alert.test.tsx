import React from 'react'
import {render} from '../../../utils'

import {Alert} from '../component/Alert'
// replace with actual path

describe('Alert component', () => {
  const testId = {
    alertTestId: 'alert-test-id',
    leftIconTestId: 'alert-left-icon',
    rightIconTestId: 'alert-right-icon',
  }

  it('renderlandığında snapshot ile eşleşmeli', () => {
    const alert = render(<Alert text='Test Alert' size='full' variant='warning' />)

    expect(alert).toMatchSnapshot()
  })

  it('renderlandığında ekranda olmalı', () => {
    const {alertTestId} = testId
    const {getByTestId} = render(<Alert text='Test Alert' size='full' variant='warning' />)

    const alertComponent = getByTestId(alertTestId)

    expect(alertComponent).toBeOnTheScreen()
  })

  it('left icon ve text alanı olmalı', () => {
    const {alertTestId, leftIconTestId} = testId
    const {getByTestId} = render(
      <Alert text='Test Alert' size='full' variant='warning' leftIcon={{name: 'WARNING'}} />
    )

    const alertComponent = getByTestId(alertTestId)
    const leftIconComponent = getByTestId(leftIconTestId)

    expect(alertComponent).toBeOnTheScreen()
    expect(leftIconComponent).toBeOnTheScreen()
  })

  it('left icon ekranda olmamalı', () => {
    const {alertTestId, leftIconTestId} = testId
    const {queryByTestId, getByTestId} = render(
      <Alert text='Test Alert' size='full' variant='warning' />
    )

    const alertComponent = getByTestId(alertTestId)
    const leftIconComponent = queryByTestId(leftIconTestId)

    expect(alertComponent).toBeOnTheScreen()
    expect(leftIconComponent).not.toBeOnTheScreen()
  })

  it('right icon ve text alanı olmalı', () => {
    const {alertTestId, rightIconTestId} = testId
    const {getByTestId} = render(
      <Alert text='Test Alert' size='full' variant='warning' rightIcon={{name: 'WARNING'}} />
    )

    const alertComponent = getByTestId(alertTestId)
    const rightIconComponent = getByTestId(rightIconTestId)

    expect(alertComponent).toBeOnTheScreen()
    expect(rightIconComponent).toBeOnTheScreen()
  })

  it('right icon ekranda olmamalı', () => {
    const {alertTestId, rightIconTestId} = testId
    const {queryByTestId, getByTestId} = render(
      <Alert text='Test Alert' size='full' variant='warning' />
    )

    const alertComponent = getByTestId(alertTestId)
    const rightIconComponent = queryByTestId(rightIconTestId)

    expect(alertComponent).toBeOnTheScreen()
    expect(rightIconComponent).not.toBeOnTheScreen()
  })

  it('left ve right icon alanları olmalı', () => {
    const {alertTestId, leftIconTestId, rightIconTestId} = testId
    const {getByTestId} = render(
      <Alert
        text='Test Alert'
        size='full'
        variant='warning'
        leftIcon={{name: 'WARNING'}}
        rightIcon={{name: 'WARNING'}}
      />
    )

    const alertComponent = getByTestId(alertTestId)
    const leftIconComponent = getByTestId(leftIconTestId)
    const rightIconComponent = getByTestId(rightIconTestId)

    expect(alertComponent).toBeOnTheScreen()
    expect(leftIconComponent).toBeOnTheScreen()
    expect(rightIconComponent).toBeOnTheScreen()
  })
})
