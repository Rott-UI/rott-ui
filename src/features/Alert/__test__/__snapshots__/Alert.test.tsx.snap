// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renderlandığında snapshot ile eşleşmeli 1`] = `
[
  <View
    alignItemsCenter={true}
    backgroundColor="rgba(255, 117, 24, 1)"
    paddingHorizontal={16}
    paddingVertical={8}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 117, 24, 1)",
        "bottom": undefined,
        "flex": undefined,
        "flexDirection": "row",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": 31,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": 16,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": "100%",
        "zIndex": undefined,
      }
    }
    testID="alert-test-id"
  >
    <Text
      height={16}
      size="sm"
      style={
        {
          "backgroundColor": undefined,
          "bottom": undefined,
          "color": "rgba(255, 255, 255, 1)",
          "flexGrow": undefined,
          "fontFamily": "Markpro-Medium",
          "fontSize": 14,
          "fontWeight": undefined,
          "gap": undefined,
          "height": 31,
          "left": undefined,
          "letterSpacing": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": 31,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "textAlign": "auto",
          "top": undefined,
          "width": 635,
          "zIndex": undefined,
        }
      }
      width={330}
    >
      Test Alert
    </Text>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
